packages/xo-server/src/xapi/mixins/patching.mjs

```diff
@@ -678,10 +678,33 @@ const methods = {
     /* throw */ notImplemented()
   },
 
+  async _updateLinstorPackages() {
+    const hosts = Object.values(this.objects.indexes.type.host)
+    const nSteps = 4
+    const nOperations = nSteps * hosts.length
+    let operationDone = 0
+
+    const callPluginOnAllHost = async (plugin, fn, args) => {
+      for (const host of hosts) {
+        await this.call('host.call_plugin', host.$ref, plugin, fn, args)
+        operationDone++
+        task.set('progress', (operationDone / nOperations) * 100)
+      }
+    }
+
+    const task = new Task({ properties: { name: 'Updating LINSTOR packages', progress: 0 } })
+    return task.run(async () => {
+      await callPluginOnAllHost('updater.py', 'update', { packages: 'xcp-ng-xapi-plugins' })
+      await callPluginOnAllHost('updater.py', 'update', { packages: 'xcp-ng-linstor' })
+      await callPluginOnAllHost('service.py', 'stop_service', { service: 'linstor-controller' })
+      await callPluginOnAllHost('service.py', 'restart_service', { service: 'linstor-satellite' })
+      task.set('progress', 100)
+    })
+  },
+
   async rollingPoolUpdate($defer, parentTask, { xsCredentials, force = false, rebootVm = force } = {}) {
-    // Temporary workaround until XCP-ng finds a way to update linstor packages
     if (some(this.objects.indexes.type.SR, { type: 'linstor' })) {
-      throw new Error('rolling pool update not possible since there is a linstor SR in the pool')
+      await this._updateLinstorPackages()
```

Comment:
will this trigger an update even if the linstorage is disconnected  ? will it be ok ?

Reply:
Yes. This has been discussed with the XCP-ng team